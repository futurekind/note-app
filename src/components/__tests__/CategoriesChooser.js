import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Chooser from '../CategoryChooser';

describe('CategoryChooser', () => {

    const wrapper = shallow(
        <Chooser />
    )

    const categories = [
        { id: 'cat1', color: 'red', label: 'Cat 1', active: true },
        { id: 'cat2', color: 'blue', label: 'Cat 2', active: false },
        { id: 'cat3', color: 'green', label: 'Cat 3', active: true },
    ]

    it('renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders list of categories', () => {
        wrapper.setProps({
            categories
        })

        const list = wrapper.childAt(0).childAt(0);

        expect(toJSON(list)).toMatchSnapshot()
    })

})