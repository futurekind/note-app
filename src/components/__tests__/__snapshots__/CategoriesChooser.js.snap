exports[`CategoryChooser renders 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Keine Kategorie
    </styled.div>
    <Styled(Icon)
      id="down"
      size={18} />
  </styled.div>
</styled.div>
`;

exports[`CategoryChooser renders list of categories 1`] = `
<styled.ul>
  <styled.li
    color="red" />
  <styled.li
    color="green" />
</styled.ul>
`;
